### ISSUE TITLE
Calling `push` on a notification instance throws an error

### ISSUE THREAD

**Issue by mshka** [jonhue/notifications-rails#4]
Hello and thank you for maintaining this gem 👋 
Trying to create and push a notification following [your instructions](https://github.com/jonhue/notifications-rails/tree/master/notification-pusher#pushers) but getting an error `NoMethodError (private method `initialize_pusher' called for #<Notification:0x00007fb23539a578>)`

### Expected behavior
it should not return an error and send a notification to the user

### Actual behavior
`NoMethodError (private method 'initialize_pusher' called for #<Notification:0x00007fb23539a578>)`

### Steps to reproduce
Pusher definition:
```ruby
NotificationPusher.configure do |config|
    config.define_pusher :ActionMailer, email: 'my@email.com'
end
```

```ruby
irb(main):001:0> notification = Notification.create target: User.first, object: Recipe.first
=> #<Notification id: 3, target_type: "User", target_id: 4, object_type: "Recipe", object_id: 316669, read: false, metadata: {}, created_at: "2018-04-09 14:51:53", updated_at: "2018-04-09 14:51:53", type: nil, subscription_id: nil, category: nil>
irb(main):002:0> notification.push :ActionMailer
Traceback (most recent call last):
        1: from (irb):2
NoMethodError (private method 'initialize_pusher' called for #<Notification:0x00007fb23539a578>)
```

### System configuration

**Ruby version**:
`ruby "2.5.0"`

**Rails version**:
`gem "rails", "5.1.4"`

**Comment by jonhue** [comment]
Please try `notification = Notification.create target: User.first, object: Recipe.first, push: :ActionMailer`

**Comment by jonhue** [comment]
@mshka Did that work?

**Comment by mshka** [comment]
@jonhue yes thank you, it worked though I think I used `pusher: :ActionMailer` instead of `push:`

**Comment by jonhue** [comment]
Still, it's not intended that calling `push` on a notification instance throws an error. I'll investigate.

**Comment by jonhue** [comment]
Fixed with version 1.2.5
